src/libraries/ObservationLib.sol

```diff
@@ -47,13 +47,13 @@ library ObservationLib {
    * @return atOrAfter Observation recorded at, or after, the target.
    */
   function binarySearch(
-    Observation[MAX_CARDINALITY] storage _observations,
+    Observation[MAX_CARDINALITY] memory _observations,
```

Comment:
Imagine the gas costs of loading the whole array into memory!  This has to be storage